### PR TITLE
Update language about minimum supported k8s version

### PR DESCRIPTION
**Pull request by mdlinville**
Also add a style guide entry about referring to CRDB versions.

## Tests
- [x] Local build
- [x] Local linkcheck 

## Reviewers
🙏 
- @meridional The k8s doc changes (only need to review the latest version)
- @nickvigilante The style guide update and the way I'm using the variables in the k8s content
- @stbof Overall content reviewer

### CODE REVIEW

**Review by nickvigilante — APPROVED** [review]
Approved in the context of docs toolchain.

**Comment by meridional** [comment]
LGTM for the k8s doc changes.

**Comment by mdlinville** [comment]
@meridional approved offline.

File: StyleGuide.md [hunk 1]
```diff
@@ -831,6 +832,34 @@ If a feature has been backported to a previous version in a patch release, use t
...
+
+**Warning**: If you use a `stable` link on a versioned page which is for a previous version, the link points to a different version  of CockroachDB than the version the page documents. Similarly, if you use a `stable` link on a page for the current version and then a new version is added, the link points to a different version than the version the page documents. If this is a problem, use one of the following methods instead.
+
+To refer to a page's minor version (for example, {{page.version.version}}), which matches its top-level subdirectory within the docs repo:
+
+~~~
+{{page.version.version}}
+~~~
+
+To refer to a page's patch version (for example, {{ page.release_info.name }}):
```
Comment (stbof) [inline StyleGuide.md:857]: Don't you want a real example of a patch version?  The version should be in backticks.

File: StyleGuide.md [hunk 2]
```diff
@@ -831,6 +832,34 @@ If a feature has been backported to a previous version in a patch release, use t
...
+
+To dynamically refer to the stable version of CockroachDB, as determined each time the site is built:
+
+~~~
+{{site.versions["stable"]}}
+~~~
+
+**Warning**: If you use a `stable` link on a versioned page which is for a previous version, the link points to a different version  of CockroachDB than the version the page documents. Similarly, if you use a `stable` link on a page for the current version and then a new version is added, the link points to a different version than the version the page documents. If this is a problem, use one of the following methods instead.
+
+To refer to a page's minor version (for example, {{page.version.version}}), which matches its top-level subdirectory within the docs repo:
```
Comment (stbof) [inline StyleGuide.md:851]: Don't you want a real example of a minor version?  The version should be in backticks.

**Review by stbof — CHANGES_REQUESTED** [review]

File: StyleGuide.md (hunk 1 above)
Comment (mdlinville) [reply StyleGuide.md]: Fixed (we synced about backticks, not needed)

File: StyleGuide.md (hunk 2 above)
Comment (mdlinville) [reply StyleGuide.md]: Fixed (we synced about backticks, not needed)

**Review by stbof — APPROVED** [review]

**Pull request merged**